var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -622,13 +624,18 @@ class Openmpi(AutotoolsPackage, CudaPackage):
     depends_on("openssh", type="run", when="+rsh")
 
     depends_on("cuda", type=("build", "link", "run"), when="@5: +cuda")
+    depends_on("hip", type=("build", "link", "run"), when="@5: +rocm")
+
+    requires("fabrics=ucx", when="+rocm")
```

Comment:
we don't want this.  although you may not be able to test it libfabric fabric can also be used with rocm.  that's how we get it to work on ORNL Frontier system.

Reply:
If you can confirm that the deployed `libfabric` at Frontier can handle this part, then we can definitely adjust this line to allow `libfabric` as well with the `one_of` syntax.

Reply:
The AMD folks want to keep that text in the open mpi documentation even though its not entirely accurate.  libfabric supports AMD accelerators using the ```FI_HMEM``` feature.  Your suggestion to use the ```one_of``` will be fine.

Reply:
Hmm, I don't see anything related to `FI_HMEM` in the `libfabric` Spack recipe... Unless it's "hidden" under `verbs` or something?

Reply:
that is something to do with libfabric not the spack view of open mpi.

Reply:
the libfabric spack recipe is probably old.  anyway, for the DOE exascale systems you'll be wanting to build against the external libfabric provided by the vendor.

Reply:
I'm tempted to put in `fabrics=cxi` there since all current DOE exascale systems use Slingshot, but that would discriminate against non-Cray/HPE customers :joy: